src/Native/Unix/System.Security.Cryptography.Native/opensslshim.h

```diff
@@ -168,16 +168,16 @@ void SSL_get0_alpn_selected(const SSL* ssl, const unsigned char** protocol, unsi
     PER_FUNCTION_BLOCK(ERR_reason_error_string, true) \
     PER_FUNCTION_BLOCK(EVP_aes_128_cbc, true) \
     PER_FUNCTION_BLOCK(EVP_aes_128_ecb, true) \
-    PER_FUNCTION_BLOCK(EVP_aes_128_gcm, true) \
-    PER_FUNCTION_BLOCK(EVP_aes_128_ccm, true) \
+    PER_FUNCTION_BLOCK(EVP_aes_128_gcm, false) \
+    PER_FUNCTION_BLOCK(EVP_aes_128_ccm, false) \
```

Comment:
Assuming i understand https://github.com/openssl/openssl/blob/OpenSSL_1_0_1-stable/util/libeay.num#L4244 correctly, the only way that EVP_aes_128_ccm is not defined is if OpenSSL was built completely without AES.

Since we don't make the other AES functions acceptable to be missing, I think that these can go back to `true`, and the `API_EXISTS` checks can be removed.

Whatever problem we're having on Ubuntu 14.04 is not related to the function not existing, otherwise it would have failed to load the module altogether.

If this is for macOS with no OpenSSL, that's a much higher problem, that will manifest much differently.

Reply:
I think I found a way make CCM work on Ubuntu 14.04 - tiny thing they require before setting the nonce/tag length - will send new iteration after fixing other comments